repose-aggregator/core/repose-core/src/test/groovy/org/openrepose/nodeservice/httpcomponent/RequestProxyServiceImplTest.groovy

```diff
@@ -161,4 +169,188 @@ class RequestProxyServiceImplTest extends Specification {
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.put("http://www.google.com", "", Collections.emptyMap(), [] as byte[])
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(null) }
+    }
+
+    def "put(uri, path, headers, body, connPoolId) will try to use a given connection pool"() {
+        given:
+        String testPoolId = "test-pool-id"
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.put("http://www.google.com", "", Collections.emptyMap(), [] as byte[], testPoolId)
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(testPoolId) }
+    }
+
+    def "patch(uri, path, headers, body) will try to use a given connection pool"() {
+        given:
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.patch("http://www.google.com", "", Collections.emptyMap(), [] as byte[])
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(null) }
+    }
+
+    def "patch(uri, path, headers, body, connPoolId) will try to use a given connection pool"() {
+        given:
+        String testPoolId = "test-pool-id"
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.patch("http://www.google.com", "", Collections.emptyMap(), [] as byte[], testPoolId)
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(testPoolId) }
+    }
+
+    def "delete(uri, path, headers) will try to use a given connection pool"() {
+        given:
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.patch("http://www.google.com", "", Collections.emptyMap())
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(null) }
+    }
+
+    def "delete(uri, path, headers, connPoolId) will try to use a given connection pool"() {
+        given:
+        String testPoolId = "test-pool-id"
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.delete("http://www.google.com", "", Collections.emptyMap(), testPoolId)
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(testPoolId) }
+    }
+
+    // TODO: Use Spock Mocking framework to avoid this hack (caused by Mockito mocks returning null)
+    def noAssertionError(Closure verification) {
```

Comment:
Well, Mockito was already being used in the test, so I was just going to continue using it.

The issue with Mockito's verify in Spock tests is that verify does not return a value; it only throws an error if the assertion is not satisfied. Well, Spock treats that as a failure. That is, `null`, `false`, and `0` are all treated as failure values by Spock in the `then` clause.

I could totally use Spock's Mocking framework, which is what the TODO I left suggests, I just didn't know if that's the path we wanted to take. I'm not sure we use Spock's Mocking framework anywhere else, and I would probably move this entire test over to it so that we aren't mixing Spock Mocks and Mockito Mocks.